runtime/common/src/precompile/stable_asset.rs

```diff
@@ -86,115 +86,140 @@ where
 			Action::GetStableAssetPoolTokens => {
 				let pool_id = input.u32_at(1)?;
 
-				let pool_info =
-					<nutsfinance_stable_asset::Pallet<Runtime> as StableAsset>::pool(pool_id).ok_or_else(|| {
-						PrecompileFailure::Revert {
-							exit_status: ExitRevert::Reverted,
-							output: "Pool not found".into(),
-							cost: target_gas_limit(target_gas).unwrap_or_default(),
+				if let Some(pool_info) = <nutsfinance_stable_asset::Pallet<Runtime> as StableAsset>::pool(pool_id) {
+					// dynamic gas cost calculation
+					// cost of reading asset currencies
+					gas_cost = gas_cost.saturating_add(
+						pool_info
+							.assets
+							.iter()
+							.map(|x| InputPricer::<Runtime>::read_currency(*x))
+							.sum::<u64>(),
+					);
+					// make sure there's enough gas
+					if let Some(gas_limit) = target_gas {
+						if gas_limit < gas_cost {
+							return Err(PrecompileFailure::Error {
+								exit_status: ExitError::OutOfGas,
+							});
 						}
```

Comment:
Could move this logic to `Pricer`, I think overlay would make it take the same amount of time. This way is fine too

Reply:
Yeah make sense, would iterate over twice